packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/IrUtils.kt

```diff
@@ -285,14 +286,26 @@ data class SchemaProperty(
     val publicName: String
 
     init {
-        if (hasPersistedNameAnnotation) {
+        val declarationName = declaration.name.identifier
+        val persistedAnnotationName: String? = if (hasPersistedNameAnnotation) getPersistedName(declaration) else null
+
+        // We only set it if the persisted and public names are different because core does
+        // would detect it as a duplicated name.
```

Comment:
```suggestion
        // We only set the public name if the persisted and public names are different 
        // because core would otherwise detect it as a duplicated name and fail.
```